test/remote_multi.py

```diff
@@ -0,0 +1,149 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys, os, subprocess, glob, argparse, pexpect
+from pexpect import pxssh
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+class Host:
+    def __init__(self):
+        self.host = ""
+        self.user = ""
+        self.password = ""
+        self.testfile = ""
+
+# @BARRO Example command:
+# > python3 remote_multi.py --hosts raspa raspe raspi raspo raspu --users test --passwords test
+# --testfiles rasp[aeiou]_test.xml
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-ds", "--discoveryserver", help="discovery-server executable")
+group = parser.add_argument_group("hosts", "list of remote machines where launch the test.")
+group.add_argument("--hosts", help="List of Hosts", required=True, action="append", nargs='+')
+group.add_argument("--users", help="List of User to login into each Host. Can be only one if all hosts share it.",
+                    required=True, action="append", nargs='+')
+group.add_argument("--passwords",
+                    help="List of Password to login as User into each Host.  Can be only one if all hosts share it.",
+                    required=True, action="append", nargs='+')
+group.add_argument("--testfiles", help="List of Test files to execute in each Host", required=True, action="append",
+                    nargs='+')
+args = parser.parse_args()
+
+hosts=[]
+for hs in args.hosts:
+    for h in hs:
+        new_host = Host()
+        new_host.host = h
+        hosts.append(new_host)
+
+i = 0
+for us in args.users:
+    if len(us) == 1:
+        for h in hosts:
+            h.user = us[0]
+    else:
+        if len(us) != len(hosts):
+            print ("ERROR: Number of users mismatch the number of hosts.")
+            sys.exit(-1)
+        for u in us:
+            hosts[i].user = u
+            i += 1
+
+i = 0
+for ps in args.passwords:
+    if len(ps) == 1:
+        for h in hosts:
+            h.password = ps[0]
+    else:
+        if len(ps) != len(hosts):
+            print ("ERROR: Number of passwords mismatch the number of hosts.")
+            sys.exit(-1)
+        for p in ps:
+            hosts[i].password = p
+            i += 1
+
+i = 0
+for ts in args.testfiles:
+    if len(ts) != len(hosts):
+        print ("ERROR: Number of test files mismatch the number of hosts.")
+        sys.exit(-1)
+    for t in ts:
+        hosts[i].testfile = t
+        i += 1
+
+# @BARRO DEBUG PRINT
+for h in hosts:
+    print(h.host)
+    print(h.user)
+    print(h.password)
+    print(h.testfile)
+    print("----")
+
+
+
+########## @BARRO REMOVE THIS sys.exit after testing to launch the test.
+sys.exit(0)
+##########
```

Comment:
Oyoyoyoyoyoy

Reply:
All multiprocess stuff Will be in a standstill till I've time enough to review it and introduce the corresponding tests

Reply:
It is a good idea to include them into this release then?